tgui/packages/tgui/interfaces/CommunicationsConsole.js

```diff
@@ -0,0 +1,712 @@
+            />
+          )}
+        </Table>
+      </Section>
+    );
+
+    if (messagingSector) {
+      children.push(<MessageModal
+        label="Message to send to allied station"
+        notice="Please be aware that this process is very expensive, and abuse will lead to...termination."
+        icon="bullhorn"
+        buttonText="Send"
+        onBack={() => setMessagingSector(null)}
+        onSubmit={message => {
+          act("sendToOtherSector", {
+            destination: messagingSector,
+            message,
+          });
+
+          setMessagingSector(null);
+        }}
+      />);
+    }
+  }
+
+  return children;
+};
+
+const PageMessages = (props, context) => {
+  const { act, data } = useBackend(context);
+  const messages = data.messages || [];
+
+  const children = [];
+
+  children.push((
+    <Section>
+      <Button
+        icon="chevron-left"
+        content="Back"
+        onClick={() => act("setState", { state: STATE_MAIN })}
+      />
+    </Section>
+  ));
+
+  const messageElements = [];
+
+  for (const [messageIndex, message] of Object.entries(messages)) {
+    let answers = null;
+
+    if (message.possibleAnswers.length > 0) {
+      answers = (
+        <Box mt={1}>
+          {message.possibleAnswers.map((answer, answerIndex) => (
+            <Button
+              content={answer}
+              color={message.answered === answerIndex + 1 ? "good" : undefined}
+              key={answerIndex}
+              onClick={message.answered ? undefined : () => act("answerMessage", {
+                message: messageIndex + 1,
+                answer: answerIndex + 1,
+              })}
+            />
+          ))}
+        </Box>
+      );
+    }
+
+    messageElements.push((
+      <Section
+        title={message.title}
+        key={messageIndex}
+        buttons={(
+          <Button.Confirm
+            icon="trash"
+            content="Delete"
+            color="red"
+            onClick={() => act("deleteMessage", {
+              message: messageIndex + 1,
+            })}
+          />
+        )}>
+        <Box>{message.content}</Box>
+
+        {answers}
+      </Section>
+    ));
+  }
+
+  children.push(messageElements.reverse());
+
+  return children;
+};
+
+export const CommunicationsConsole = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    authenticated,
+    authorizeName,
+    canLogOut,
+    emagged,
+    page,
+  } = data;
+
+  let pageComponent = null;
+
+  if (authenticated) {
+    switch (page) {
+      case STATE_BUYING_SHUTTLE:
+        pageComponent = <PageBuyingShuttle />;
+        break;
+      case STATE_CHANGING_STATUS:
+        pageComponent = <PageChangingStatus />;
+        break;
+      case STATE_MAIN:
+        pageComponent = <PageMain />;
+        break;
+      case STATE_MESSAGES:
+        pageComponent = <PageMessages />;
+        break;
+      default:
+        pageComponent = <Box>Page not implemented: {page}</Box>;
+    }
+  }
```

Comment:
Again, suggesting a slightly terser syntax, switch cases with reassignment are pretty long and ugly.

```suggestion
  const pageComponent = !!authenticated && (
    page === STATE_BUYING_SHUTTLE && <PageBuyingShuttle />
    || page === STATE_CHANGING_STATUS && <PageChangingStatus />
    || page === STATE_MAIN && <PageMain />
    || page === STATE_MESSAGES && <PageMessages />
    || <Box>Page not implemented: {page}</Box>
  );
```

Reply:
Is this common syntax in JS? As someone unfamiliar with it the logic you're suggesting looks weird and I find it difficult to parse even after seeing the equivalent switch case structure